sdk/lib/rtl/critical.c

```diff
@@ -575,62 +583,95 @@ RtlInitializeCriticalSectionAndSpinCount(PRTL_CRITICAL_SECTION CriticalSection,
     CriticalSection->SpinCount = (NtCurrentPeb()->NumberOfProcessors > 1) ? SpinCount : 0;
     CriticalSection->LockSemaphore = 0;
 
-    /* Allocate the Debug Data */
-    CritcalSectionDebugData = RtlpAllocateDebugInfo();
-    DPRINT("Allocated Debug Data: %p inside Process: %p\n",
-           CritcalSectionDebugData,
-           NtCurrentTeb()->ClientId.UniqueProcess);
-
-    if (!CritcalSectionDebugData)
+    if (Flags & RTL_CRITICAL_SECTION_FLAG_NO_DEBUG_INFO)
     {
-        /* This is bad! */
-        DPRINT1("Couldn't allocate Debug Data for: %p\n", CriticalSection);
-        return STATUS_NO_MEMORY;
+        CriticalSection->DebugInfo = LongToPtr(-1);
```

Comment:
I wonder why DebugInfo set to NULL is _not enough_ to signal that the critical section indeed doesn't have debug information....

(Side remark: if you early-return success there, this would allow to now reindent the whole existing code below, thus reducing the overall diff.)

Reply:
Guess #1: Is there some way to delay allocate the debug info?

Guess #2: Compatibility thing because the CS internals have changed a couple of times. Maybe some apps initialize directly with {0} instead of calling InitializeCriticalSection?

Reply:
I added a test. That test shows it is like that.